support/helper.py

```diff
@@ -141,63 +160,216 @@ def dict_reverse_cumsum(d, col, sum_col):
+
+        if k1 == 'checking':
+            df1 = aggregate_dict_by_month(data, {'amount': 'sum'})
+            metadata[k1][k2][k3]['monthly'] = {}
+            metadata[k1][k2][k3]['monthly']['total_count'] = len(df1)
+            metadata[k1][k2][k3]['monthly']['balance'] = df1['amount'].tolist()
+            metadata[k1][k2][k3]['monthly']['overdraft_count'] = len(df1[df1['amount'] < 0].index)
+
+    metadata[k1][k2][k3]['current'] = current
+    metadata[k1][k2][k3]['limit'] = limit
+    metadata[k1][k2][k3]['high_balance'] = high_balance
+
+    # running balance
+    data = pd.DataFrame(lst)
+    data['date'] = pd.to_datetime(data['date'], format='%Y-%m-%d')
+    temp = data.groupby([data['date'].dt.year, data['date'].dt.month], as_index=False).last()
+    metadata[k1][k2][k3]['running_balance'] = temp['current'].tolist()
+
+    # credit card util ratio
     if df is not None:
         metadata = util_ratio(metadata, df)
-    return metadata
 
+    # transactions
+    k3 = 'transactions'
+    metadata[k1][k2][k3] = {}
+    metadata[k1][k2][k3]['total_count'] = len(lst)
+    metadata[k1][k2][k3]['timespan'] = lst[0]['timespan']
+
+    m = metadata[k1][k2][k3]['timespan'] / 30
+    metadata[k1][k2][k3]['avg_monthly_count'] = metadata[k1][k2][k3]['total_count'] / m
+    metadata[k1][k2][k3]['avg_monthly_value'] = sum(d['amount'] for d in lst) / m
 
-def transactions(metadata, lst, key):
-    ''' regardless how many different account within same account type '''
-    metadata[key]['transactions'] = {}
-    metadata[key]['transactions']['total_count'] = len(lst)
-    metadata[key]['transactions']['timespan'] = lst[0]['timespan']
-    timespan_in_months = metadata[key]['transactions']['timespan'] / 30
-    metadata[key]['transactions']['avg_monthly_count'] = metadata[key]['transactions']['total_count'] / timespan_in_months
-    metadata[key]['transactions']['avg_monthly_cash_flow'] = sum(d['amount']*-1 for d in lst) / timespan_in_months
     return metadata
 
 
 def late_payment(metadata, lst):
     metadata['credit_card']['late_payment'] = {}
+    metadata['credit_card']['late_payment']['general'] = {}
+    metadata['credit_card']['late_payment']['period'] = {}
     period = [30, 60, 90, 180, 360, 720, 1800]
-    data = [d for d in lst if d['category'] == 'interest']
-    for p in period:
-        metadata['credit_card']['late_payment'][p] = len([d for d in data if d['timespan'] <= p])
+    data = [d for d in lst if d['sub_category'] == 'interest charged']
+    if data:
+        for p in period:
+            metadata['credit_card']['late_payment']['period'][p] = len([d for d in data if d['timespan'] <= p])
+        metadata['credit_card']['late_payment']['general']['total_count'] = len(data)
+        metadata['credit_card']['late_payment']['general']['month_count'] = len(
+            aggregate_dict_by_month(data, {'amount': ['count']}))
+    return metadata
+
+
+def income(metadata, lst, key, value, k1='checking', k2='income'):
```

Comment:
Same here: we might need docstrings for the functions:
`keep_dict`, `expenses`, `investments`, `cash_flow`, `earnings`.